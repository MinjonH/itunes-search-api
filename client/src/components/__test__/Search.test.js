import { render, screen } from '@testing-library/react';
import Search from '../search.js';
import renderer from 'react-test-renderer';
import optionsList from '../search.js';


it('should render same text passed into button', () => {
    render(<Search />);
    const buttonElement = screen.getByRole("button", {name: 'Search'});
    expect(buttonElement).toBeInTheDocument();
});

it('should render all option properties', () => {
    const optionElement = renderer.create(<Search/>).toJSON();
    expect(optionElement).toMatchSnapshot();
});
    
