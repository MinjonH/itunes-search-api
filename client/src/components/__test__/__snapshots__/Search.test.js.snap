// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all option properties 1`] = `
<div
  className="SearchForm"
>
  <h1
    className="SearchHeading"
  >
    Search iTunes
  </h1>
  <form>
    <div
      className="frmInput"
    >
      <input
        name="term"
        placeholder="Search..."
        type="text"
      />
      <select
        name="media"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="audiobook"
        >
          Audio Book
        </option>
        <option
          value="ebook"
        >
          e-Book
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="musicVideo"
        >
          Music Video
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
      </select>
      <button>
        Search
      </button>
    </div>
  </form>
</div>
`;
